BDArmory/Weapons/Missiles/ModuleMissileRearm.cs

```diff
@@ -66,6 +79,50 @@ public bool SpawnMissile(Transform MissileTransform, float offset = 0, bool dedu
             return false;
         }
 
+        public void loadOrdinance(int tubesToReload)
+        {
+            if (BDArmorySettings.DEBUG_MISSILES) Debug.Log($"[BDArmory.ModuleMissileRearm] reloading {tubesToReload} launchrails, {railAmmo} ordinance in launcher, queuing {tubesToReload - railAmmo} new munitions from magazine");
+            if (linkedMagazines.Count > 0 && ammoCount > 0 && railAmmo < tubesToReload) //no/not enough ammo internal to launcher? grab some from magazine
+            {
+                int neededReloads = (int)(tubesToReload - railAmmo);
+                for (int t2r = 0; t2r < neededReloads; t2r++)
+                {
+                    ModuleMissileMagazine priorityMagazine = null;
+                    float lastPriority = -1;
+                    float lastAmmoQty = -1;
+                    for (int mmm = 0; mmm < linkedMagazines.Count; mmm++)
+                    {
+                        if (linkedMagazines[mmm].ammoCount >= 1)
+                        {
+                            if (linkedMagazines[mmm].priority >= lastPriority)
+                            {
+                                lastPriority = linkedMagazines[mmm].priority;
+                                if (linkedMagazines[mmm].ammoCount >= lastAmmoQty) //FIXME - sometimes priority is ignored, revise logic later
```

Comment:
```csharp
foreach (var mag in linkedMagazines)
{
    if (mag == null || mag.ammoCount < 1) continue; // Ignore broken or empty mags.
    if (mag.priority < lastPriority) continue; // Ignore lower priority mags.
    if (mag.priority > lastPriority)
    {
        lastAmmoQty = -1; // Reset the ammo quantity, so that the quantity of lower priority mags aren't considered.
        lastPriority = mag.priority;
    }
    if (mag.ammoCount < lastAmmoQty) continue; // Ignore mags with the same priority but less ammo.
    lastAmmoQty = mag.ammoCount;
    priorityMagazine = mag;
}
```
should do what you want here.